tests/lib/prepare.sh

```diff
@@ -572,22 +659,52 @@ EOF
     # - built debs
     # - golang archive files and built packages dir
     # - govendor .cache directory and the binary,
-    rsync -a -C \
+    if is_core16_system || is_core18_system; then
+        rsync -a -C \
           --exclude '*.a' \
           --exclude '*.deb' \
           --exclude /gopath/.cache/ \
           --exclude /gopath/bin/govendor \
           --exclude /gopath/pkg/ \
           /home/gopath /mnt/user-data/
+    elif is_core20_system; then
+        # prepare passwd for wormhole
+        mkdir -p /root/test-etc
+        mkdir -p /var/lib/extrausers
+        touch /var/lib/extrausers/sub{uid,gid}
+        for f in group gshadow passwd shadow; do
+            grep -v "^root:" "/etc/$f" > /root/test-etc/"$f"
+            grep "^root:" /etc/"$f" >> /root/test-etc/"$f"
+            chgrp --reference "/etc/$f" /root/test-etc/"$f"
+            # create /var/lib/extrausers/$f
+            # append ubuntu, test user for the testing
+            grep "^test:" /etc/$f >> /var/lib/extrausers/"$f"
+            grep "^ubuntu:" /etc/$f >> /var/lib/extrausers/"$f"
+            # check test was copied
+            MATCH "^test:" </var/lib/extrausers/"$f"
```

Comment:
what's the reasoning behind needing to check that these things were added properly?

Reply:
Just to ensure that there are no nasty surprises, if this code does not add users (e.g. because of a shell typo/thinko somewhere) the system will just sit there and not much happens. It's hard to debug so adding these extra checks ensures at least this part is correct :)